### PR TITLE
faster/efficient `EmoteExtractor` and `Helpers`

### PR DESCRIPTION
**Pull request by AoshiW**
# Benchmarks
## Helpers.ParseBadges
Before:
| Method |               badges |     Mean |   Error |  StdDev |   Gen0 | Allocated |
|------- |--------------------- |---------:|--------:|--------:|-------:|----------:|
|    One |        subscriber/12 | 101.9 ns | 1.97 ns | 2.89 ns | 0.0484 |     304 B |
|    Two| subsc(...)ter/1 [27] | 188.6 ns | 2.58 ns | 3.86 ns | 0.0980 |     616 B |
|  theree | subsc(...)rbo/1 [35] | 235.9 ns | 2.42 ns | 3.47 ns | 0.1259 |     792 B |
|   Four| subsc(...)dsa/1 [47] | 293.8 ns | 3.04 ns | 4.55 ns | 0.1564 |     984 B |

After:
| Method |               badges |      Mean |    Error |   StdDev |   Gen0 | Allocated |
|------- |--------------------- |----------:|---------:|---------:|-------:|----------:|
|    One |        subscriber/12 |  87.95 ns | 2.353 ns | 3.299 ns | 0.0370 |     232 B |
|    Two| subsc(...)ter/1 [27] | 121.25 ns | 1.150 ns | 1.650 ns | 0.0484 |     304 B |
| theree| subsc(...)rbo/1 [35] | 161.52 ns | 2.241 ns | 3.285 ns | 0.0572 |     360 B |
|    Four | subsc(...)dsa/1 [47] | 190.27 ns | 2.767 ns | 4.056 ns | 0.0675 |     424 B |

## EmoteExtractor 
Before:
|         Method |           emoteSetStr |              message |       Mean |   Error |   StdDev |   Gen0 |   Gen1 | Allocated |
|--------------- |--------------------- |--------------------- |-----------:|--------:|---------:|-------:|-------:|----------:|
| MultipleEmotes | 25:0-(...)82-92 [65] | Kappa(...)heese [93] | 1,195.6 ns | 9.91 ns | 14.83 ns | 0.6771 | 0.0038 |    4248 B |
|       OneEmote |             25:25-29 | Twitc(...)Kappa [30] |   204.8 ns | 6.15 ns |  8.82 ns | 0.1159 |      - |     728 B |

After:
|         Method |           emoteSetStr |              message |     Mean |   Error |  StdDev |   Gen0 |   Gen1 | Allocated |
|--------------- |--------------------- |--------------------- |---------:|--------:|--------:|-------:|-------:|----------:|
| MultipleEmotes | 25:0-(...)82-92 [65] | Kappa(...)heese [93] | 650.9 ns | 5.92 ns | 7.91 ns | 0.3595 | 0.0019 |    2256 B |
|       OneEmote |             25:25-29 | Twitc(...)Kappa [30] | 111.7 ns | 1.28 ns | 1.87 ns | 0.0612 |      - |     384 B |

### CODE REVIEW

**Comment by neon-sunset** [comment]
Banger. Out of curiousity, what are the numbers before/after? You can add a new benchmark if you want to https://github.com/TwitchLib/TwitchLib.Client/tree/dev/TwitchLib.Client.Benchmark

File: TwitchLib.Client.Models/Extensions/SpanSliceEnumerator.cs
Comment (neon-sunset) [inline TwitchLib.Client.Models/Extensions/SpanSliceEnumerator.cs:1]: You can also add `SpanExtensions` (or `SplitExtensions`) with `public SpanSliceEnumerator<T> Split<T>(this ReadOnlySpan<T> source, T value) where T : IEquatable<T>` to make it more convenient.

Comment (AoshiW) [reply TwitchLib.Client.Models/Extensions/SpanSliceEnumerator.cs:1]: I thought about it, but I decided not to add it.

**Comment by AoshiW** [comment]
benchmark result added

**Review by neon-sunset — APPROVED** [review]

**Review by Syzuna — APPROVED** [review]

**Comment by Syzuna** [comment]
LGTM thanks!

**Pull request merged**